src/main/java/core/basesyntax/service/impl/CsvReportServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ReportingService;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class CsvReportServiceImpl implements ReportingService {
+    public static final String CSV_SEPARATOR = ",";
+    private String header;
+
+    public CsvReportServiceImpl(String header) {
+        this.header = header;
+    }
+
+    @Override
+    public List<String> generateReport() {
+        List<String> report = new ArrayList<>(Storage.storage.size() + 1);
+        report.add(header);
+        report.addAll(getDataFromStorageInKeyValueView());
+        return report;
+    }
```

Comment:
It's a good idea just to return a simple String value here. I believe it would not take as much memory as ArrayList